File: javaproject/app/src/main/java/com/marcosvbras/robomarket/viewmodels/BaseViewModel.java
package com.marcosvbras.robomarket.viewmodels;

import android.arch.lifecycle.ViewModel;
import android.databinding.ObservableBoolean;
import android.databinding.ObservableField;

import com.marcosvbras.robomarket.interfaces.BaseActivityCallback;

public class BaseViewModel extends ViewModel {

    public ObservableField<String> toolbarTitle = new ObservableField<>();
    public ObservableBoolean isLoading = new ObservableBoolean(false);
    public BaseActivityCallback activityCallback;

}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/viewmodels/BaseViewModel.kt
package com.marcosvbras.robomarket.viewmodels

import android.arch.lifecycle.ViewModel
import android.databinding.ObservableBoolean
import android.databinding.ObservableField
import com.marcosvbras.robomarket.interfaces.ActivityCallback

abstract class BaseViewModel : ViewModel() {

    val toolbarTitle = ObservableField<String>()
    val isLoading = ObservableBoolean(false)
    val activityCallback: ActivityCallback? = null

    abstract fun cleanupSubscriptions()

}